pipeline/backend/kubernetes/pod.go

```diff
@@ -146,6 +150,9 @@ func podContainer(step *types.Step, podName, goos string) (v1.Container, error)
 		return container, err
 	}
 
+	if (step.Type == types.StepTypeService) && (!storageRwx) {
+		return container, nil
```

Comment:
`container` returns on line 161.
Gathering volume mounts (lines 156-159) should be wrapped in that `if`.